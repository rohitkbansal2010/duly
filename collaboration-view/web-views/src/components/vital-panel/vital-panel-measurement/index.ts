export { VitalPanelMeasurement } from './vital-panel-measurement';
export type { VitalPanelMeasurementPropsType } from './vital-panel-measurement';
