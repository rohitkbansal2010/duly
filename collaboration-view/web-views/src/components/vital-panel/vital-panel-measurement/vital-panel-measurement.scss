@import "~@styles/main.scss";
@import "~@styles/colors.scss";
@import "~@styles/typography.scss";

.vitalPanelMeasurementContainer {
  display: flex;
  align-items: center;
  padding: 1.25rem 0 1.25rem 1rem;

  &:not(:first-child) {
    border-left: 0.06rem solid $iron;
  }

  &Icon {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 3.75rem;
    height: 3.75rem;
    margin-right: 1rem;
    border-radius: 50%;
    @include background-opacity($dupage-magenta, 0.3);

    & img {
      object-fit: cover;
    }

    &BloodPressure {
      width: 2.4rem;
      height: 2.02rem;
    }

    &HeartRate {
      width: 2.29rem;
      height: 2.1rem;
    }

    &Temperature {
      width: 1.38rem;
      height: 2.5rem;
    }

    &RespiratoryRate {
      width: 2.29rem;
      height: 2.29rem;
    }

    &PainLevel {
      width: 1.16rem;
      height: 2.06rem;
    }
  }

  &AdditionalInfo {
    &TopBlock,
    &BottomBlock {
      display: flex;
    }

    &TopBlock {
      align-items: center;
    }

    &BottomBlock {
      align-items: baseline;
    }

    &Text {
      color: $medium-gray;
      font-size: var(--font-size-16px-4px);
      line-height: 1.5rem;
      @include nexa-heavy-highlight;
    }

    &Date {
      margin-left: 0.75rem;
      color: $dupage-violet;
      font-size: var(--font-size-14px-4px);
      line-height: 1rem;
      @include nexa-extra-bold-highlight;
    }

    &Value {
      min-height: 2.625rem;
      color: $dupage-magenta;
      font-size: var(--font-size-32px-4px);
      line-height: 2.625rem;
      @include nexa-heavy-highlight;
    }

    &Unit,
    &MaxScaleValue,
    &AdditionalValue {
      margin-left: 0.25rem;
      font-size: var(--font-size-22px-4px);
      line-height: 1.625rem;
      @include nexa-heavy-highlight;
    }

    &Unit,
    &MaxScaleValue {
      color: $dupage-magenta;
    }

    &AdditionalValue {
      color: $dupage-violet;
    }
  }
}
