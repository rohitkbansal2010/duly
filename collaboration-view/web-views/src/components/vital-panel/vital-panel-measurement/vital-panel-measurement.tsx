import { CardType } from '@enums';
import { formatMDYYYYDate } from '@utils';

import { MeasurementComponentType } from '../config-vital-panels';

import styles from './vital-panel-measurement.scss';

export type VitalPanelMeasurementPropsType = MeasurementComponentType & {
  cardType: CardType;
  isVisible: boolean;
  convertedHeightUnit: string | null;
  convertedWeightUnit: string | null
  convertedHeightValue?: string | null;
  convertedWeightValue?: number | null;
  toggle: string;
};

export const VitalPanelMeasurement = ({
  cardType,
  isVisible,
  text,
  value,
  convertedWeightValue,
  convertedHeightValue,
  convertedHeightUnit,
  convertedWeightUnit,
  style,
  unit,
  maxScaleValue,
  additionalValue,
  icon,
  date,
}: VitalPanelMeasurementPropsType) => {
  if (!isVisible) return null;
  
  return (
    
    
    <div className={styles.vitalPanelMeasurementContainer} style={style}>     
      {icon && (
        <div className={styles.vitalPanelMeasurementContainerIcon}>
          <img
            src={icon}
            alt="vital icon"
            className={styles[`vitalPanelMeasurementContainerIcon${cardType}`]}
            date-test="vital-panel-measurement-icon"
          />
        </div>
      )}
      <div className={styles.vitalPanelMeasurementContainerAdditionalInfo}>
        <div className={styles.vitalPanelMeasurementContainerAdditionalInfoTopBlock}>
          <div
            className={styles.vitalPanelMeasurementContainerAdditionalInfoText}
            date-test="vital-panel-measurement-text"
          >
            {text}
          </div>
          {date && (
            <div
              className={styles.vitalPanelMeasurementContainerAdditionalInfoDate}
              date-test="vital-panel-measurement-date"
            >
              {formatMDYYYYDate(new Date(date))}
            </div>
          )}
        </div>
        <div className={styles.vitalPanelMeasurementContainerAdditionalInfoBottomBlock}>
          <div
            className={styles.vitalPanelMeasurementContainerAdditionalInfoValue}
            date-test="vital-panel-measurement-value"
          >
            {text === 'WEIGHT' && convertedWeightValue ? convertedWeightValue : 
            text === 'HEIGHT' && convertedHeightValue ? convertedHeightValue : value}
          </div>
          {!additionalValue && unit && (
            <div
              className={styles.vitalPanelMeasurementContainerAdditionalInfoUnit}
              date-test="vital-panel-measurement-unit"
            >
              {text === 'WEIGHT' && convertedWeightUnit ? convertedWeightUnit : 
            text === 'HEIGHT' && convertedHeightUnit ? '' : unit}
            </div>
          )}
          {maxScaleValue && (
            <div
              className={styles.vitalPanelMeasurementContainerAdditionalInfoMaxScaleValue}
              date-test="vital-panel-measurement-max-scale-value"
            >
              {`/ ${maxScaleValue}`}
            </div>
          )}
          {additionalValue && (
            <>
              <div
                className={styles.vitalPanelMeasurementContainerAdditionalInfoAdditionalValue}
                date-test="vital-panel-measurement-additional-value"
              >
                {`/ ${additionalValue}`}
              </div>
              <div className={styles.vitalPanelMeasurementContainerAdditionalInfoUnit}>
                {unit}
              </div>
            </>
          )}
        </div>
      </div>
    </div>
  );
};
