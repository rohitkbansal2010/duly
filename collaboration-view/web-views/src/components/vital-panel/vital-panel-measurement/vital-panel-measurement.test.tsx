import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { CardType } from '@enums';
import { conditionMagentaIcon } from '@icons';
import { formatMDYYYYDate } from '@utils';

import { VitalPanelMeasurement, VitalPanelMeasurementPropsType } from './vital-panel-measurement';

const mockProps: VitalPanelMeasurementPropsType = {
  cardType: CardType.BLOOD_PRESSURE,
  isVisible: true,
  text: 'MOST RECENTLY',
  value: 130,
  style: { width: '44%' },
  unit: 'mmHg',
  maxScaleValue: 10,
  additionalValue: 80,
  icon: conditionMagentaIcon,
  date: '2021-12-01T11:52:37.5832382+00:00',
};

const setUp = (props: VitalPanelMeasurementPropsType) =>
  shallow(<VitalPanelMeasurement {...props} />);

describe('VitalPanelMeasurement', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    component = setUp(mockProps);
  });

  it('should render correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should not render the VitalPanelMeasurement component, if the isVisible flag be falsy', () => {
    const component = setUp({ ...mockProps, isVisible: false });

    expect(toJson(component)).toMatchSnapshot();
    expect(component.type()).toBeNull();
  });

  it('should render the icon', () => {
    const iconElement = component.find('[date-test="vital-panel-measurement-icon"]');

    expect(iconElement).toHaveLength(1);
    expect(iconElement.props().src?.includes(mockProps.icon as string)).toBeTruthy();
  });

  it('should not render the icon, if the icon does not exist', () => {
    const component = setUp({ ...mockProps, icon: '' });

    expect(toJson(component)).toMatchSnapshot();

    const iconElement = component.find('[date-test="vital-panel-measurement-icon"]');

    expect(iconElement).toHaveLength(0);
  });

  it('should render the text', () => {
    const textElement = component.find('[date-test="vital-panel-measurement-text"]');

    expect(textElement).toHaveLength(1);
    expect(textElement.text().includes(mockProps.text as string)).toBeTruthy();
  });

  it('should render the date', () => {
    const dateElement = component.find('[date-test="vital-panel-measurement-date"]');
    const expectedDate = formatMDYYYYDate(new Date(mockProps.date as string));

    expect(dateElement).toHaveLength(1);
    expect(dateElement.text().includes(expectedDate)).toBeTruthy();
  });

  it('should not render the date, if the date does not exist', () => {
    const component = setUp({ ...mockProps, date: '' });

    expect(toJson(component)).toMatchSnapshot();

    const dateElement = component.find('[date-test="vital-panel-measurement-date"]');

    expect(dateElement).toHaveLength(0);
  });

  it('should render the value', () => {
    const valueElement = component.find('[date-test="vital-panel-measurement-value"]');

    expect(valueElement).toHaveLength(1);
    expect(valueElement.text().includes(String(mockProps.value))).toBeTruthy();
  });

  it('should render the maxScaleValue', () => {
    const maxScaleValueElement = component.find('[date-test="vital-panel-measurement-max-scale-value"]');
    const expectedMaxScaleValue = `/ ${mockProps.maxScaleValue}`;

    expect(maxScaleValueElement).toHaveLength(1);
    expect(maxScaleValueElement.text().includes(expectedMaxScaleValue)).toBeTruthy();
  });

  it('should not render the maxScaleValue, if the maxScaleValue does not exist', () => {
    const component = setUp({ ...mockProps, maxScaleValue: undefined });

    expect(toJson(component)).toMatchSnapshot();

    const maxScaleValueElement = component.find('[date-test="vital-panel-measurement-max-scale-value"]');

    expect(maxScaleValueElement).toHaveLength(0);
  });

  it('should render the unit', () => {
    const component = setUp({ ...mockProps, additionalValue: undefined });

    expect(toJson(component)).toMatchSnapshot();

    const unitElement = component.find('[date-test="vital-panel-measurement-unit"]');

    expect(unitElement).toHaveLength(1);
    expect(unitElement.text().includes(mockProps.unit as string)).toBeTruthy();
  });

  it('should not render the unit, if the unit does not exist', () => {
    const component = setUp({ ...mockProps, unit: '' });

    expect(toJson(component)).toMatchSnapshot();

    const unitElement = component.find('[date-test="vital-panel-measurement-unit"]');

    expect(unitElement).toHaveLength(0);
  });

  it('should render the additionalValue', () => {
    const additionalValueElement = component.find('[date-test="vital-panel-measurement-additional-value"]');
    const expectedAdditionalValue = `/ ${mockProps.additionalValue}`;

    expect(additionalValueElement).toHaveLength(1);
    expect(additionalValueElement.text().includes(expectedAdditionalValue)).toBeTruthy();
  });

  it('should not render the additionalValue, if the additionalValue does not exist', () => {
    const component = setUp({ ...mockProps, additionalValue: undefined });

    expect(toJson(component)).toMatchSnapshot();

    const additionalValueElement = component.find('[date-test="vital-panel-measurement-additional-value"]');

    expect(additionalValueElement).toHaveLength(0);
  });
});
