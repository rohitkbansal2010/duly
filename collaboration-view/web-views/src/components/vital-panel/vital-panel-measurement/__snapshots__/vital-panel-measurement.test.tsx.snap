// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VitalPanelMeasurement should not render the VitalPanelMeasurement component, if the isVisible flag be falsy 1`] = `""`;

exports[`VitalPanelMeasurement should not render the additionalValue, if the additionalValue does not exist 1`] = `
<div
  className="vitalPanelMeasurementContainer"
  style={
    Object {
      "width": "44%",
    }
  }
>
  <div
    className="vitalPanelMeasurementContainerIcon"
  >
    <img
      alt="vital icon"
      className="vitalPanelMeasurementContainerIconBloodPressure"
      date-test="vital-panel-measurement-icon"
      src="condition-magenta.svg"
    />
  </div>
  <div
    className="vitalPanelMeasurementContainerAdditionalInfo"
  >
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoTopBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoText"
        date-test="vital-panel-measurement-text"
      >
        MOST RECENTLY
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoDate"
        date-test="vital-panel-measurement-date"
      >
        12/1/2021
      </div>
    </div>
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoBottomBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoValue"
        date-test="vital-panel-measurement-value"
      >
        130
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoUnit"
        date-test="vital-panel-measurement-unit"
      >
        mmHg
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoMaxScaleValue"
        date-test="vital-panel-measurement-max-scale-value"
      >
        / 10
      </div>
    </div>
  </div>
</div>
`;

exports[`VitalPanelMeasurement should not render the date, if the date does not exist 1`] = `
<div
  className="vitalPanelMeasurementContainer"
  style={
    Object {
      "width": "44%",
    }
  }
>
  <div
    className="vitalPanelMeasurementContainerIcon"
  >
    <img
      alt="vital icon"
      className="vitalPanelMeasurementContainerIconBloodPressure"
      date-test="vital-panel-measurement-icon"
      src="condition-magenta.svg"
    />
  </div>
  <div
    className="vitalPanelMeasurementContainerAdditionalInfo"
  >
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoTopBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoText"
        date-test="vital-panel-measurement-text"
      >
        MOST RECENTLY
      </div>
    </div>
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoBottomBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoValue"
        date-test="vital-panel-measurement-value"
      >
        130
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoMaxScaleValue"
        date-test="vital-panel-measurement-max-scale-value"
      >
        / 10
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoAdditionalValue"
        date-test="vital-panel-measurement-additional-value"
      >
        / 80
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoUnit"
      >
        mmHg
      </div>
    </div>
  </div>
</div>
`;

exports[`VitalPanelMeasurement should not render the icon, if the icon does not exist 1`] = `
<div
  className="vitalPanelMeasurementContainer"
  style={
    Object {
      "width": "44%",
    }
  }
>
  <div
    className="vitalPanelMeasurementContainerAdditionalInfo"
  >
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoTopBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoText"
        date-test="vital-panel-measurement-text"
      >
        MOST RECENTLY
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoDate"
        date-test="vital-panel-measurement-date"
      >
        12/1/2021
      </div>
    </div>
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoBottomBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoValue"
        date-test="vital-panel-measurement-value"
      >
        130
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoMaxScaleValue"
        date-test="vital-panel-measurement-max-scale-value"
      >
        / 10
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoAdditionalValue"
        date-test="vital-panel-measurement-additional-value"
      >
        / 80
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoUnit"
      >
        mmHg
      </div>
    </div>
  </div>
</div>
`;

exports[`VitalPanelMeasurement should not render the maxScaleValue, if the maxScaleValue does not exist 1`] = `
<div
  className="vitalPanelMeasurementContainer"
  style={
    Object {
      "width": "44%",
    }
  }
>
  <div
    className="vitalPanelMeasurementContainerIcon"
  >
    <img
      alt="vital icon"
      className="vitalPanelMeasurementContainerIconBloodPressure"
      date-test="vital-panel-measurement-icon"
      src="condition-magenta.svg"
    />
  </div>
  <div
    className="vitalPanelMeasurementContainerAdditionalInfo"
  >
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoTopBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoText"
        date-test="vital-panel-measurement-text"
      >
        MOST RECENTLY
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoDate"
        date-test="vital-panel-measurement-date"
      >
        12/1/2021
      </div>
    </div>
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoBottomBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoValue"
        date-test="vital-panel-measurement-value"
      >
        130
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoAdditionalValue"
        date-test="vital-panel-measurement-additional-value"
      >
        / 80
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoUnit"
      >
        mmHg
      </div>
    </div>
  </div>
</div>
`;

exports[`VitalPanelMeasurement should not render the unit, if the unit does not exist 1`] = `
<div
  className="vitalPanelMeasurementContainer"
  style={
    Object {
      "width": "44%",
    }
  }
>
  <div
    className="vitalPanelMeasurementContainerIcon"
  >
    <img
      alt="vital icon"
      className="vitalPanelMeasurementContainerIconBloodPressure"
      date-test="vital-panel-measurement-icon"
      src="condition-magenta.svg"
    />
  </div>
  <div
    className="vitalPanelMeasurementContainerAdditionalInfo"
  >
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoTopBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoText"
        date-test="vital-panel-measurement-text"
      >
        MOST RECENTLY
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoDate"
        date-test="vital-panel-measurement-date"
      >
        12/1/2021
      </div>
    </div>
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoBottomBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoValue"
        date-test="vital-panel-measurement-value"
      >
        130
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoMaxScaleValue"
        date-test="vital-panel-measurement-max-scale-value"
      >
        / 10
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoAdditionalValue"
        date-test="vital-panel-measurement-additional-value"
      >
        / 80
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoUnit"
      />
    </div>
  </div>
</div>
`;

exports[`VitalPanelMeasurement should render correctly 1`] = `
<div
  className="vitalPanelMeasurementContainer"
  style={
    Object {
      "width": "44%",
    }
  }
>
  <div
    className="vitalPanelMeasurementContainerIcon"
  >
    <img
      alt="vital icon"
      className="vitalPanelMeasurementContainerIconBloodPressure"
      date-test="vital-panel-measurement-icon"
      src="condition-magenta.svg"
    />
  </div>
  <div
    className="vitalPanelMeasurementContainerAdditionalInfo"
  >
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoTopBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoText"
        date-test="vital-panel-measurement-text"
      >
        MOST RECENTLY
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoDate"
        date-test="vital-panel-measurement-date"
      >
        12/1/2021
      </div>
    </div>
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoBottomBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoValue"
        date-test="vital-panel-measurement-value"
      >
        130
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoMaxScaleValue"
        date-test="vital-panel-measurement-max-scale-value"
      >
        / 10
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoAdditionalValue"
        date-test="vital-panel-measurement-additional-value"
      >
        / 80
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoUnit"
      >
        mmHg
      </div>
    </div>
  </div>
</div>
`;

exports[`VitalPanelMeasurement should render the unit 1`] = `
<div
  className="vitalPanelMeasurementContainer"
  style={
    Object {
      "width": "44%",
    }
  }
>
  <div
    className="vitalPanelMeasurementContainerIcon"
  >
    <img
      alt="vital icon"
      className="vitalPanelMeasurementContainerIconBloodPressure"
      date-test="vital-panel-measurement-icon"
      src="condition-magenta.svg"
    />
  </div>
  <div
    className="vitalPanelMeasurementContainerAdditionalInfo"
  >
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoTopBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoText"
        date-test="vital-panel-measurement-text"
      >
        MOST RECENTLY
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoDate"
        date-test="vital-panel-measurement-date"
      >
        12/1/2021
      </div>
    </div>
    <div
      className="vitalPanelMeasurementContainerAdditionalInfoBottomBlock"
    >
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoValue"
        date-test="vital-panel-measurement-value"
      >
        130
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoUnit"
        date-test="vital-panel-measurement-unit"
      >
        mmHg
      </div>
      <div
        className="vitalPanelMeasurementContainerAdditionalInfoMaxScaleValue"
        date-test="vital-panel-measurement-max-scale-value"
      >
        / 10
      </div>
    </div>
  </div>
</div>
`;
